src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -292,20 +337,79 @@ void java_bytecode_convert_classt::convert(const classt &c)
     throw 0;
   }
 
-  // now do fields
+  // Now do fields
+  for(auto overlay_class : overlay_classes)
+  {
+    for(const auto &field : overlay_class.get().fields)
+    {
+      debug()
+        << "Adding symbol from overlay class:  field '" << field.name << "'"
+        << eom;
+      convert(*class_symbol, field);
+    }
+  }
   for(const auto &field : c.fields)
   {
     debug() << "Adding symbol:  field '" << field.name << "'" << eom;
     convert(*class_symbol, field);
   }
 
-  // now do methods
-  for(const auto &method : c.methods)
+  // Now do methods
+  std::set<irep_idt> overlay_methods;
+  for(auto overlay_class : overlay_classes)
+  {
+    for(const methodt &method : overlay_class.get().methods)
+    {
+      const irep_idt method_identifier =
+        id2string(qualified_classname) + "." + id2string(method.name)
+          + ":" + method.descriptor;
+      if(method_bytecode.contains_method(method_identifier))
+      {
+        if(overlay_methods.count(method_identifier) == 0)
+        {
+          warning()
+            << "Method " << method_identifier
+            << " exists in an overlay class without being marked as an "
+              "overlay and also exists in another overlay class that appears "
+              "earlier in the classpath"
+            << eom;
+        }
+        continue;
+      }
+      // Always run the lazy pre-stage, as it symbol-table
+      // registers the function.
+      debug()
+        << "Adding symbol from overlay class:  method '" << method_identifier
+        << "'" << eom;
+      java_bytecode_convert_method_lazy(
+        *class_symbol,
+        method_identifier,
+        method,
+        symbol_table,
+        get_message_handler());
+      method_bytecode.add(qualified_classname, method_identifier, method);
+      if(is_overlay_method(method))
+        overlay_methods.insert(method_identifier);
+    }
+  }
+  for(const methodt &method : c.methods)
   {
     const irep_idt method_identifier=
       id2string(qualified_classname)+
       "."+id2string(method.name)+
       ":"+method.descriptor;
+    if(method_bytecode.contains_method(method_identifier))
```

Comment:
This is false when we've added a overlay method from a class? Perhaps a comment here